prime-router/src/main/kotlin/cli/ProcessFhirCommands.kt

```diff
@@ -186,76 +211,145 @@ class ProcessFhirCommands : CliktCommand(
+                    (receiver != null && receiver.format == MimeFormat.FHIR)
+                ) -> {
+                return convertFhirToFhir(FhirTranscoder.encode(bundle), receiver, senderSchema, isCli)
             }
 
             // HL7 to FHIR to HL7 conversion
             inputFileType == "HL7" && (
                 outputFormat == MimeFormat.HL7.toString() ||
-                (receiver != null && (receiver.format == MimeFormat.HL7 || receiver.format == MimeFormat.HL7_BATCH))
-            ) -> {
+                    (
+                        receiver != null &&
+                        (receiver.format == MimeFormat.HL7 || receiver.format == MimeFormat.HL7_BATCH)
+                    )
+                ) -> {
                 val (bundle2, inputMessage) = convertHl7ToFhir(contents, receiver)
-                val output = convertFhirToHl7(FhirTranscoder.encode(bundle2))
-                outputResult(output)
-                if (diffHl7Output != null) {
-                    val differences = hl7DiffHelper.diffHl7(output, inputMessage)
+                val output = convertFhirToHl7(
+                    jsonString = FhirTranscoder.encode(bundle2),
+                    senderSchema = senderSchema,
+                    isCli = isCli
+                )
+                if (diffHl7Output != null && isCli) {
+                    val differences = hl7DiffHelper.diffHl7(output.message!!, inputMessage)
                     echo("-------diff output")
                     echo("There were ${differences.size} differences between the input and output")
                     differences.forEach { echo(it.toString()) }
                 }
+                return output
             }
 
             else -> throw CliktError("File extension ${inputFile.extension} is not supported.")
         }
+        return messageOrBundle
     }
 
+    private fun setEnrichmentSchemaFields(
+        messageOrBundle: MessageOrBundle,
+        enrichmentSchemaFields: FhirTransformer.BundleWithMessages,
+    ): MessageOrBundle {
+        messageOrBundle.enrichmentSchemaWarnings.addAll(enrichmentSchemaFields.warnings)
+        messageOrBundle.enrichmentSchemaErrors.addAll(enrichmentSchemaFields.errors)
+        messageOrBundle.enrichmentSchemaPassed = enrichmentSchemaFields.errors.isEmpty()
+        messageOrBundle.bundle = enrichmentSchemaFields.bundle
+        return messageOrBundle
+    }
+
+    private fun setEnrichmentSchemaFields(
+        messageOrBundle: MessageOrBundle,
+        warnings: MutableList<String>,
+        errors: MutableList<String>,
+        bundle: Bundle,
+    ): MessageOrBundle {
+        messageOrBundle.enrichmentSchemaWarnings.addAll(warnings)
+        messageOrBundle.enrichmentSchemaErrors.addAll(errors)
+        messageOrBundle.enrichmentSchemaPassed = errors.isEmpty()
+        messageOrBundle.bundle = bundle
+        return messageOrBundle
+    }
+
+    class MessageOrBundle(
```

Comment:
Why do we need to classes with almost the same properties? Can we create a base class?